src/graphql/__tests__/mutations.js

```diff
@@ -0,0 +1,88 @@
+'use strict' 
+
+const EasyGraphQLTester = require('easygraphql-tester')
+const fs = require('fs')
+const path = require('path')
+
+const schema = fs.readFileSync(path.join(__dirname, '..', 'schema.graphql'), 'utf8')
+
+describe('Test GraphQL mutations', () => {
+  let tester
+  
+  beforeAll(() => {
+    tester = new EasyGraphQLTester(schema)
+  })
+
+  describe('Mutations', () => {
+    test('Should pass a valid createContributor', () => {
+      const mutation = `
+        mutation CREATE_CONTRIBUTOR($input: CreateContributorInput) {
+          createContributor(input: $input) {
+            githubUsername
+            email
+            github {
+              username
+              contributionCount
+              pullRequests {
+                id
+                title
+                url
+                number
+                labels {
+                  name
+                  url
+                }
+              }
+            }
+            shopify {
+              id
+              codes {
+                code
+                used
+              }
+            }
+          }
+        }
+      `
+      tester.test(true, mutation, {
+        githubUsername: 'demo',
+        email: 'demo@demo.com',
+        firstName: 'demo',
+        acceptsMarketing: true
+      })
+    })
```

Comment:
Does this execute the queries? Meaning does the underlying data access happen?

Reply:
And if not, how does this work?

Reply:
This will not execute any query/mutation, what it does under the hood is that it first parse your schema, then parse the query/mutation, then create a mock of the query/mutation but in the process of creating the query/mutation it is validating that everything is valid, I mean, it's gonna check that the query/mutation exists, then that the require fields exist on the type, that the arguments, input types exist and are valid.... and much more... also, you can use instead of  `tester.test(true, query)` -> `tester.mock(query)` and it'll return a mock of your query/mutations there you can set fixtures as well!
```
const query = `
  {
    getUser(id: "1") {
      id
      name
      familyInfo {
        lastName
        email
      }
    }
  }
`

const fixture = {
  id: '1',
  name: 'EasyGraphQL'
}

const { getUser } = tester.mock({ query, fixture })

// getUser
{ 
  id: '1',
  name: 'EasyGraphQL',
  familyInfo: [
    { 
      lastName: 'Bartoletti',
      email: 'YSjsYuV@wtnK.com'
    },
    { 
      lastName: 'Bartoletti',
      email: 'YSjsYuV@wtnK.com'
    },
    { 
      lastName: 'Bartoletti',
      email: 'YSjsYuV@wtnK.com'
    }
  ]
}
```

Reply:
Actually it’s being mocked from the [schema](https://github.com/gatsbyjs/api.gatsbyjs.org/blob/master/src/graphql/schema.graphql#L54), the fields has some types, so, easygraphql will go through The types and will mock each field inside it!! It’s not testing the resolvers, it’s testing the schema with the queries and mutations!

Reply:
Gotcha. So in that case, wouldn't this be better suited for a _consumer_ of a GraphQL API? Otherwise, we're basically adding a chore to write down an (unused) full GraphQL query that's tested against the schema, which is kind of the same as testing whether or not GraphQL's type system works.

I can see huge value in this in a UI, though, if all queries were extracted and tested against the schema to find any breaking changes.

However, in an API scenario, this seems like duplicating built-in type-safety checks.

Reply:
This will be a great option too, using it on the consumer of the GraphQL API is a good way to find if there is going to be any breaking change!!

But also, using it on the API scenario is a good way to know if everything is going to work as expected while creating the schema, I mean, for example, changing a field on one input type should break the resolvers too, but also, just testing the resolvers will have no effect, because it is going to handle the previous value(mock) but it’s not going to break with the new value...

Reply:
[Here](https://youtu.be/2INxv2EMuVE) is quick video that explain what I mean